lib/active_record/virtual_attributes/virtual_delegates.rb

```diff
@@ -15,6 +15,8 @@ module ClassMethods
         #
 
         def virtual_delegate(*methods, to:, type:, prefix: nil, allow_nil: nil, default: nil, uses: nil, **options) # rubocop:disable Naming/MethodParameterName
+          ActiveRecord::VirtualAttributes.deprecator.warn("virtual_delegate is now deprecated. use virtual_attribute :through instead", caller_locations)
+
```

Comment:
GitHub won't let me comment below but on lines 33-38 below should this method be changed to go through `virtual_attribute :through`? I thikn you can drop the `define_delegate` line and the bit to `raise ArgumentError, "Delegation...` because those are already in virtual_attribute :through.

Reply:
(Can be in a follow up PR - I just noticed the duplication)

Reply:
I didn't want to work too hard for this one.
The goal is to drop this method all together.

I'm not sure exactly how to invert, I'll look into this.
But do we want to spend the $$$ to do this?

Reply:
Doesn't really matter to me - I was trying to reduce the duplication, but yeah if it's going away quickly, then it really doesn't matter.

Reply:
We tend to not specify `uses`.

- With `arel`: 4 of the 45 specify `uses`.
- Without `arel`: 157 of the 371 specify `uses`.
- With delegate: 2 of the 48 specify `uses`.

Without `arel` manually traverse associations, so I expected it to be a higher percentage. Delegates traverse another object, so I wondered if that would be zero. 

The use case we are looking at is the delegate. Since 5% use the `to` and not the `uses`, I'd prefer to keep the `(uses || to)` in there.

The 2 delegate with uses that are similar: `:to => :hardware, :uses => {:hardware => :disks}`. Fun fact: The `:uses` was not necessary a few years ago. We introduced it when we stopped introspecting the target class. (The "Constant" problem race condition that Joe fixed)